libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.cpp

```diff
@@ -130,151 +152,260 @@ bool File::open (char const *filepath, uint8_t mode)
     char const* splash = strrchr(filepath, '/');
     strncpy(_name, splash ? (splash + 1) : filepath, LFS_NAME_MAX);
   }
-
   return ret;
 }
 
 size_t File::write (uint8_t ch)
 {
-  VERIFY(!_is_dir, 0);
   return write(&ch, 1);
 }
 
 size_t File::write (uint8_t const *buf, size_t size)
 {
-  VERIFY(!_is_dir, 0);
-
-  lfs_ssize_t wrcount = lfs_file_write(_fs->getFS(), _file, buf, size);
-  VERIFY(wrcount > 0, 0);
+  lfs_ssize_t wrcount = 0;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
```

Comment:
this should be checked before attempting to lock the fs

Reply:
I disagree.  The same `File` object can change from a file to a directory, such as through a call to `File::open(char const *filepath, uint8_t mode)`.  Therefore, it is necessary to lock before checking the member variable `_is_dir`.

Reply:
IMHO, it is over-engineer,  I don't see use case where user change want to change it to other filepath in another thread out of context.